scripts/tests/CRM/test_1b_decrypt_fidelity.py

```
@@ -63,31 +63,70 @@ def inspect_raw_database(db_path: str, user_id: int) -> dict:
-    repo_root = SCRIPT_DIR.parent.parent.parent
-
-    # Get user record with -json flag for structured output
-    # Use shlex.quote for db_path to handle paths with spaces/special chars
-    cmd = f"docker compose exec -T backend sqlite3 -json {shlex.quote(db_path)} 'SELECT * FROM users WHERE id = {user_id}'"
-    result = subprocess.run(cmd, shell=True, capture_output=True, text=True, cwd=repo_root)
+    # Get user record using the hardened run_docker_sql() helper
+    try:
+        user_output = run_docker_sql(
+            f"SELECT id, pubkey, email, encrypted_email, ephemeral_pubkey_email, "
+            f"name, encrypted_name, ephemeral_pubkey_name, email_blind_index, "
+            f"user_type_id, approved, created_at FROM users WHERE id = {user_id}",
```

Ruff (0.14.14)

[error] Possible SQL injection vector through string-based query construction

(S608)